pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -126,11 +126,19 @@ func (pt PipelineTask) Validate(ctx context.Context) (errs *apis.FieldError) {
 	errs = errs.Also(pt.validateRefOrSpec())
 
 	errs = errs.Also(pt.validateEmbeddedOrType())
-
+	notCustomTaskKinds := map[TaskKind]bool{
+		"":                 true,
+		NamespacedTaskKind: true,
+		"ClusterTask":      true,
+	}
 	// Pipeline task having taskRef/taskSpec with APIVersion is classified as custom task
 	switch {
+	case pt.TaskRef != nil && !notCustomTaskKinds[pt.TaskRef.Kind]:
+		errs = errs.Also(pt.validateCustomTask())
 	case pt.TaskRef != nil && pt.TaskRef.APIVersion != "":
```

Comment:
This might be more readable as:
```
case pt.TaskRef != nil:
  if pt.TaskRef.APIVersion != "" || !notCustomTaskKinds[pt.TaskRef.Kind] {
    pt.validateCustomTask()
  }
```

(Also could the double negative !notCustomTaskKinds be made into a positive?)

Reply:
It's not so easy to merge these 2, there are some cases in this switch control, it seems not add readability with this